web/v1/snowflake/sql-runner/sql/standard/00-setup/01-main/01-stored-procedures.sql

```diff
@@ -0,0 +1,426 @@
+
+      dmls.forEach(function(stmt) {
+          snowflake.createStatement({sqlText: stmt}).execute();
+      });
+      snowflake.createStatement({sqlText: `COMMIT;`}).execute();
+
+  } catch(ERROR) {
+      snowflake.createStatement({sqlText: `ROLLBACK;`}).execute();
+
+      // Snowflake error is not very helpful here
+      var err_msg = "Transaction rolled back.Probably failed: " + DML_STATEMENTS + " :Error: ";
+      throw Error(err_msg + ERROR);
+  }
+
+  return "ok. Statements in transaction succeeded."
+
+  $$
+;
+
+
+/*
+  COLUMN_CHECK
+  Checks for mismatched columns between source and target tables.
+  If source table is missing columns, it errors.
+  If AUTOMIGRATE is 'TRUE',
+    it allows the target table to get added the additional columns,if any, in source.
+  Since ALTER TABLE is a DDL statement, it will be its own transaction.
+  This means that even if it was placed inside another transaction,
+    it would commit that, and then start another(implicit) for its own execution.
+  So, also it cannot be explicitly rolled back.
+  This is why column_check is not part of the commit_with_metadata procedure.
+
+  Input:
+  SRC_SCHEMA:  the schema of the source table
+  SRC_TBL:     the source table name
+  TRG_SCHEMA:  the schema of the target table
+  TRG_TBL:     the target table name
+  AUTOMIGRATE: whether any extra columns in source table will be added in target table (only 'TRUE' enables)
+
+  To drop:
+  DROP PROCEDURE {{.output_schema}}.column_check(VARCHAR,VARCHAR,VARCHAR,VARCHAR,VARCHAR);
+*/
+CREATE OR REPLACE PROCEDURE {{.output_schema}}.column_check(SRC_SCHEMA  VARCHAR,
+                                                            SRC_TBL     VARCHAR,
+                                                            TRG_SCHEMA  VARCHAR,
+                                                            TRG_TBL     VARCHAR,
+                                                            AUTOMIGRATE VARCHAR)
+  RETURNS VARCHAR
+  LANGUAGE JAVASCRIPT
+  AS
+  $$
+
+  var delim = '~';
+  var sourceColumns = list_cols_with_type(SRC_SCHEMA,SRC_TBL,delim).split(delim);
+  var targetColumns = list_cols_with_type(TRG_SCHEMA,TRG_TBL,delim).split(delim);
+
+  if (targetColumns.some(notIncludedIn(sourceColumns)) === true) {
+
+      throw "ERROR: Source table is missing column(s) which exist in target table.";
+
+  } else {
+
+      var columnAdditions = sourceColumns.filter(notIncludedIn(targetColumns));
+
+      if ( columnAdditions.length !== 0 ) {
+          if ( AUTOMIGRATE !== 'TRUE' ) {
+              throw "ERROR: Target table is missing column(s),but automigrate is not enabled.";
+          } else {
+              add_columns_to(TRG_SCHEMA, TRG_TBL, columnAdditions.join(','));
+              return "ok.Columns added."
+          }
+      } else {
+          return "ok. Columns match."
+      }
+  }
+
+  // == Helpers ==
+
+  function list_cols_with_type(sch,tbl,delimiter) {
+      var stmt = `
+          SELECT
+              LISTAGG(
```

Comment:
It might make sense to return an array, and LISTAGG later. I'm not tied to it though, whatever solution works best.